tests/baselines/reference/declarationEmitCrossFileCopiedGeneratedImportType.js

```diff
@@ -0,0 +1,32 @@
+//// [tests/cases/compiler/declarationEmitCrossFileCopiedGeneratedImportType.ts] ////
+
+//// [index.d.ts]
+export declare class Foo extends Number {
+    private _;
+}
+//// [index.d.ts]
+import { Foo } from "../projA";
+export declare const f: (foo: Foo) => boolean;
+//// [index.d.ts]
+export declare const e: {
+    f: (foo: import("../projA").Foo) => boolean;
+};
+//// [index.ts]
+import {e} from "../projC";
+
+export const d = {e};
+
+//// [index.js]
+"use strict";
+Object.defineProperty(exports, "__esModule", { value: true });
+exports.d = void 0;
+var projC_1 = require("../projC");
+exports.d = { e: projC_1.e };
+
+
+//// [index.d.ts]
+export declare const d: {
+    e: {
+        f: (foo: import("../projA").Foo) => boolean;

```

Comment:
If I didn't also swap to `getSpecifierForModuleSymbol` as part of this, this emits as `"../projA/index"` because, well, `getResolvedExternalModuleName` is not the _robust_ specifier generation solution in the checker that calls into `moduleSpecifiers.ts`, while `getSpecifierForModuleSymbol` is. `getResolvedExternalModuleName` was probably sufficient for just handling the unbundled -> bundled specifier case (which is all the `declarations.ts` version of `rewriteModuleSpecifier` needs to handle and what this was originally based on), but doesn't handle rewriting specifiers copied from other files, which is an additional case the node builder needs to handle (and as of this week, happens waaaaaaay more often). `getExternalModuleNameFromDeclaration` does.